### PR TITLE
🐛Change the return type of `getNode` to be nullable.

### LINKED ISSUE #280: src/main-thread/nodes.ts#getNode() should have nullable return type
Given a bogus ID, the main-thread's `getNode()` can return `null`. Therefore, its return type should be nullable i.e. `RenderableElement | null`. 

This will require downstream changes. 

Alternative: Retain the non-nullable return type, but throw a runtime exception if the `id` can't be found.

/to @delima02 /cc @kristoferbaxter

### PR DESCRIPTION
**Pull request by delima02**
* Change the return type of `src/main-thread/nodes.ts#getNode()` to be nullable.
* In places where this method is called, expect a nullable return value.
* Fixes #280

🐛
@choumx @jridgewell

### CODE REVIEW

File: src/main-thread/nodes.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export function createNode(skeleton: TransferrableNode, sanitizer?: Sanitizer):
 /**
  * Returns the real DOM Element corresponding to a serialized Element object.
  * @param id
- * @return RenderableElement
+ * @return RenderableElement | null
```
Comment (choumx) [inline src/main-thread/nodes.ts]: This only changes documentation. In TypeScript, you want to change the function type: https://www.typescriptlang.org/docs/handbook/functions.html

Once you change it, the compiler should show several type errors caused by the change. Grab me for a chat when you get to that point.

File: src/main-thread/mutator.ts [hunk 2]
```diff
@@ -42,7 +42,14 @@ const mutators: {
   [key: number]: (mutation: TransferrableMutationRecord, target: Node) => void;
 } = {
   [MutationRecordType.CHILD_LIST](mutation: TransferrableMutationRecord, target: HTMLElement) {
-    (mutation[TransferrableKeys.removedNodes] || []).forEach(node => getNode(node[TransferrableKeys.index]).remove());
+    (mutation[TransferrableKeys.removedNodes] || []).forEach(node => {
+      const retrieved = getNode(node[TransferrableKeys.index]);
```
Comment (choumx) [inline src/main-thread/mutator.ts]: s/retrieved/node

File: src/main-thread/mutator.ts [hunk 3]
```diff
@@ -42,7 +42,14 @@ const mutators: {
...
+      if (!retrieved) {
+        console.error('ERROR: getNode() yields a null value. Node id was not found.');
```
Comment (choumx) [inline src/main-thread/mutator.ts]: "ERROR" is redundant. Also, include the missing ID in the message.

File: src/main-thread/commands/bounding-client-rect.ts [hunk 4]
```diff
@@ -31,6 +31,7 @@ export function process(worker: Worker, mutation: TransferrableMutationRecord):
   const target = getNode(mutation[TransferrableKeys.target]);
 
   if (!target) {
+    console.error('ERROR: getNode() yields a null value. Node id was not found.');
```
Comment (choumx) [inline src/main-thread/commands/bounding-client-rect.ts]: Are `console.error` stripped from the production build?

File: src/main-thread/commands/event-subscription.ts [hunk 5]
```diff
@@ -134,6 +134,7 @@ export function process(worker: Worker, mutation: TransferrableMutationRecord):
   const target = getNode(mutation[TransferrableKeys.target]);
 
   if (!target) {
+    console.error('ERROR: getNode() yields a null value. Node id was not found.');
```
Comment (choumx) [inline src/main-thread/commands/event-subscription.ts]: There's a similar issue here with amphtml -- duplicate error strings can cause bundle size bloat. 

@kristoferbaxter Do you have a preference here?

File: src/main-thread/mutator.ts (hunk 3 above)
Comment (delima02) [reply src/main-thread/mutator.ts]: Done!

File: src/main-thread/mutator.ts (hunk 2 above)
Comment (delima02) [reply src/main-thread/mutator.ts]: I don't think I understand this comment. Can you give me a hand?

Comment (choumx) [reply src/main-thread/mutator.ts]: Replace "retrieved" with "node". It's a reference from `sed`: https://thoughtbot.com/blog/sed-102-replace-in-place

Comment (delima02) [reply src/main-thread/mutator.ts]: "node" already exists in this scope. Maybe "retrievedNode" or something specific?

Comment (choumx) [reply src/main-thread/mutator.ts]: I'd rename the existing `node` to `removed` or `nodeReference`.

**Comment by choumx** [comment]
Did we check if `console.error` is stripped from the production bundle?

File: src/main-thread/commands/event-subscription.ts (hunk 5 above)
Comment (kristoferbaxter) [reply src/main-thread/commands/event-subscription.ts]: If we think the errors are important, I'll add a build step that moves them to codes and generates a json file with mappings for code -> string. We can host this like we're planning with AMPHTML.

Comment (choumx) [reply src/main-thread/commands/event-subscription.ts]: They are stripped from the prod build but preserved on the dev build. 

I think this is good to go until we have an error code mapping system. Thanks Kris.

**Pull request merged**